### ISSUE TITLE
Keep getting nan for loss.

### ISSUE THREAD

**Issue by AdarshMJ** [facebookresearch/hgnn#8]
Hi thank you for this excellent paper. I am trying to run your code on the zinc task with the following command -

python -m torch.distributed.launch --nproc_per_node=1 main.py --task zinc --world_size 1 --local_rank 0 --embed_manifold hyperbolic

I checked that I keep getting this error only if I change the embed_manifold to hyperbolic. Seems to work fine for euclidean.

I keep getting this error. Any suggestions on how to solve this?

<img width="731" alt="Screenshot 2020-01-24 at 8 04 53 AM" src="https://user-images.githubusercontent.com/24205200/73040606-35f1da80-3e80-11ea-9ec3-3d79a4e6d975.png">

**Comment by AdarshMJ** [comment]
I had one more query. I was hoping you could please provide me some insights - 
The results reported in the paper, are they computed by using embed_manifold hyperbolic embeddings + Lorentz,Poincare and Euclidean or embed_manifold is kept as Euclidean and then the manifold is changed as Lorentz, Poincare etc?

**Comment by leuchine** [comment]
Hi:

Thanks a lot for your issues. Could you please provide the versions of PyTorch, NumPy, etc, that you are using? I did not encounter this problem with my machines. BTW, the results are obtained by setting the embed_manifold = euclidean and select_manifold = euclidean|poincare|lorentz. I recommend setting the embed_manifold = euclidean as it gets better results empirically. Thanks!

Best Regards,
Qi

**Comment by AdarshMJ** [comment]
I have installed from requirements.txt Im thinking it installs the latest versions?
 I am running the task zinc and this is the result its printing - 
<img width="1013" alt="Screenshot 2020-01-24 at 4 38 43 PM" src="https://user-images.githubusercontent.com/24205200/73064815-0cf53800-3ec8-11ea-8391-d4f536edb702.png">


How exactly is the results reported in the table printed? Like there are 3 properties that are reported. But the whole training is defined by just one parameter - loss

**Comment by leuchine** [comment]
Hi:

Yes. The installation shall be correct. I am not sure where the NAN comes from. Is it for Lorentz or Poincare or both? (Usually, it happens due to some numerical problems of the manifolds, which take time to debug) For the metrics, please refer to the loss, which is MAE for regression tasks. Thanks!

Best Regards,
Qi

**Comment by AdarshMJ** [comment]
Hi!
 I have only tried it with Lorentz. The loss that gets printed what should be the trend? It keeps getting smaller. Will the final results be similar to the one mentioned in Table 2 of your paper?

What I mean is, will the final result after running the task ZINC will be in the form
logP = 2.4
QED = 4.7 and SAS = 7.7 ?

**Comment by leuchine** [comment]
Hi:

Yes. As it is mean absolute error, it will get smaller. It depends on the prop_idx you chose. For zinc, prop_idx = 0|1|2, which corresponds to logP, QED and SAS, respectively. Thanks!

**Comment by AdarshMJ** [comment]
Hi!
Thank you for your patience :) The default epochs to run is for 300?

Any plans on pushing an update that allows for resuming the training? Colab keeps disconnecting :(

**Comment by leuchine** [comment]
Hi:

We used early stopping and 300 is the maximum number of epochs. So, it won't run for 300 epochs usually. 

Thanks for proposing the checkpoints. We may consider adding checkpoints later. Thanks!

**Comment by AdarshMJ** [comment]
Thank you so much!
